### PR TITLE
[MFA-2681] add Auth0TelemetryInfo to track app version usage via sdk

### PR DESCRIPTION
**Pull request by andy317fe301f8c7**
### Description

SDK should allow for an additional telemetry info that will be sent in requests to mfa-api. If not provided, a default value is set.

### References

https://auth0team.atlassian.net/browse/MFA-2681

### Testing
- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not the default branch

### CODE REVIEW

**Review by shadow-dahm — APPROVED** [review]

**Review by shadow-dahm — CHANGES_REQUESTED** [review]
Approve pending the snyk results

File: GuardianTests/Networking/ClientInfoSpec.swift [hunk 1]
```diff
@@ -60,9 +60,18 @@ class ClientInfoSpec: QuickSpec {
             }
 
             it("should have the correct header value") {
-                expect(try? clientInfo.asHeader().values).to(contain("eyJuYW1lIjoiR3VhcmRpYW4uc3dpZnQiLCJ2ZXJzaW9uIjoiMS4wLjAifQ"))
-            }
+                let decoder = JSONDecoder()
 
+                let value = try? clientInfo.asHeader().values.first
```
Comment (diegose) [inline GuardianTests/Networking/ClientInfoSpec.swift:65]: I see we are not looking at values other than the first one. Should we?

File: Guardian/Networking/ClientInfo.swift [hunk 2]
```diff
@@ -29,15 +29,17 @@ private let infoEntryName = "CFBundleShortVersionString"
 struct ClientInfo: Codable, Equatable {
     let name: String
     let version: String
+    let env: Auth0TelemetryInfo?
```
Comment (diegose) [inline Guardian/Networking/ClientInfo.swift:32]: I don't see any references to `env` outside of this struct. What is its purpose?

Comment (oleksandrozerov-okta) [reply Guardian/Networking/ClientInfo.swift:32]: this is an addition for serialized ClientInfo (`base64(json(ClientInfo))`) which is getting put in the header value, datawarehouse can give us statistics on values inside which is exactly what we aim for here

**Review by shadow-dahm — APPROVED** [review]

**Pull request merged**